app/components/github-connect.js

```diff
@@ -0,0 +1,37 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  computed,
+  get
+} = Ember;
+
+/**
+  The github-connect component is used to to connect to github
+
+  ## default usage
+
+  ```handlebars
+    {{github-connect clientId=clientId scope=scope state=state redirectUri=redirectUri}}
+  ```
+
+  @module Component
+  @class github-connect
+  @extends Ember.Component
+  @public
+ */
+
+export default Component.extend({
+  tagName: 'a',
+  classNames: ['button', 'default'],
+  attributeBindings: ['url:href'],
+
+  clientId: null,
+  scope: null,
+  state: null,
+  redirectUri: null,
+
+  url: computed('clientId', 'scope', 'state', 'redirectUrl', function() {
+    return `https://github.com/login/oauth/authorize?scope=${get(this, 'scope')}&client_id=${get(this, 'clientId')}&state=${get(this, 'state')}&redirect_uri=${get(this, 'redirectUri')}`;
```

Comment:
The URL itself is something that's not going to change and should not vary across different parts of the app, so config doesn't really help us. It would be cleaner and clearer to declare it as a `const`, though, at the top of the modules, outside the definition of the component.